packages/core/src/scss/components/checkout/customer/_customer.scss

```diff
@@ -92,19 +92,156 @@
     width: 100%;
 }
 
-.wallet-buttons-container {
-    margin: spacing('single') 0 spacing('half');
+// Button Width Function
+//
+// -----------------------------------------------------------------------------
+$buttonWidthMap: (
+    single:                     span $wallet-button-track-size,
+    half:                       span $wallet-button-track-size / 2,
+    third:                      span $wallet-button-track-size / 3,
+    quarter:                    span $wallet-button-track-size / 4
+);
+
+@function buttonWidth($key) {
+
+    @if map-has-key($buttonWidthMap, $key) {
+        @return map-get($buttonWidthMap, $key);
+    }
+
+    @warn "Unknown `#{$key}` in $buttonWidthMap.";
+    @return null;
+}
+
+// Top Wallet Buttons Container
+//
+// -----------------------------------------------------------------------------
+
+.checkout-buttons-container {
+    margin: spacing('double') 0 0;
 
     .checkoutRemote {
+        display: grid;
+        grid-gap: $floating-form-field-spacing;
+        grid-template-columns: repeat($wallet-button-track-size, 1fr);
         margin: 0;
+
+        > div {
+            height: $wallet-button-height;
+            margin-right: 0;
+        }
+
+        .AmazonPayContainer,
+        > #applepayCheckoutButton button,
+        button.gpay-button,
+        > #braintreepaypalcreditCheckoutButton,
+        > #braintreepaypalCheckoutButton {
```

Comment:
Avoid nesting and overly specific selectors if possible, see https://github.com/bigcommerce/sass-style-guide#nesting and https://github.com/bigcommerce/sass-style-guide#specificity